node_modules/col-whiteboards/lib/api.js

```diff
@@ -0,0 +1,771 @@
+      notifyOnlineUsers(whiteboardSession.whiteboard_id);
+
+      return callback();
+    });
+  });
+};
+
+/**
+ * Delete all whiteboard sessions. This is necessary when the server restarts, as all
+ * websockets will have disconnect without the disconnect logic being executed
+ *
+ * @param  {Function}       [callback]                      Standard callback function
+ * @param  {Object}         [callback.err]                  An error that occurred, if any
+ * @api private
+ */
+var deleteWhiteboardSessions = function(callback) {
+  callback = callback || function() {};
+
+  var options = {
+    'truncate': true
+  };
+
+  DB.WhiteboardSession.destroy(options).complete(function(err) {
+    if (err) {
+      log.error({'err': err}, 'Failed to delete all whiteboard session');
+      return callback({'code': 500, 'msg': err.message});
+    }
+
+    return callback();
+  });
+};
+
+// Delete all whiteboard sessions when the server starts
+deleteWhiteboardSessions();
+
+/* WHITEBOARD ELEMENTS */
+
+/**
+ * Store a new element was added to the whiteboard canvas and inform
+ * the online users in the whiteboard of the new element
+ *
+ * @param  {Socket}         socket                          The socket on which the new whiteboard element was sent
+ * @param  {Object}         element                         The deserialized whiteboard element that was added
+ * @param  {Function}       [callback]                      Standard callback function
+ * @param  {Object}         [callback.err]                  An error that occurred, if any
+ * @api private
+ */
+var addWhiteboardElement = function(socket, element, callback) {
+  callback = callback || function() {};
+
+  // Notify all online users in the whiteboard that a new whiteboard element was added.
+  // This will not send a notification over the socket on which the new element was sent
+  socket.broadcast.to(socket.whiteboard.id).emit('addElement', element);
+
+  // Store the new element
+  upsertWhiteboardElement(socket, element, callback);
+};
+
+/**
+ * Store an updated whiteboard canvas element and inform
+ * the online users in the whiteboard of the element update
+ *
+ * @param  {Socket}         socket                          The socket on which the whiteboard element update was sent
+ * @param  {Object}         element                         The deserialized updated whiteboard element
+ * @param  {Function}       [callback]                      Standard callback function
+ * @param  {Object}         [callback.err]                  An error that occurred, if any
+ * @api private
+ */
+var updateWhiteboardElement = function(socket, element, callback) {
+  callback = callback || function() {};
+
+  // Notify all online users in the whiteboard that a whiteboard element was updated.
+  // This will not send a notification over the socket on which the element update was sent
+  socket.broadcast.to(socket.whiteboard.id).emit('updateElement', element);
```

Comment:
The thinking was that we want to make the update available to the other users in the whiteboard as soon as possible. However, given that this should only be a matter of ms, I'll change it to broadcast in the callback.